engine.cpp

```diff
@@ -426,9 +426,10 @@ bool Engine::Texture_Load(const std::string& id, char *filename){
 
     createNewTexture(id,surface->w, surface->h);
 
-	glTexImage2D( GL_TEXTURE_2D, 0, nOfColors, surface->w, surface->h, 0,
+	  glTexImage2D( GL_TEXTURE_2D, 0, nOfColors, surface->w, surface->h, 0,
                       texture_format, GL_UNSIGNED_BYTE, surface->pixels );
-
+    // assuming the texture was copied to OpenGL, the surface is no longer needed
```

Comment:
Awesome comment as it answers a potential question I'll have when reading this code later on (remember, code is read more often than it's written)